src/main/java/com/epam/esm/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,74 @@
+package com.epam.esm.dao.impl;
+
+import com.epam.esm.dao.UserDao;
+import com.epam.esm.dao.request.UserSearchCriteria;
+import com.epam.esm.dao.service.PersistenceService;
+import com.epam.esm.model.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import javax.annotation.PostConstruct;
+import javax.persistence.NoResultException;
+import java.util.List;
+
+@Repository
+public class UserDaoImpl implements UserDao {
```

Comment:
Read about dao vs repository

Reply:
I read. But I think that this annotation can be put here. This is an extract from the description of this annotation: 
*Teams implementing traditional Java EE patterns such as "Data Access Object" may also apply this stereotype to DAO classes, though care should be taken to understand the distinction between Data Access Object and DDD-style repositories before doing so. This annotation is a general-purpose stereotype and individual teams may narrow their semantics and use as appropriate.*